### PR TITLE
csi: bump csi snapshotter image to v5

### PR DESCRIPTION
**Pull request by sathieu**
**Description of your changes:**

Bump csi snapshotter image to v5.0.1.

Breaking changes (from [5.0.1](https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v5.0.1) release notes:

> Replaces many VolumeSnapshot/VolumeSnapshotContent Update/UpdateStatus operations with Patch. This lowers the probability of the "object has been modified" update API errors occurring. This change introduces a dependency on two new RBAC rules for the CSI snapshotter sidecar: volumesnapshotcontents:patch, volumesnapshotcontents/status:patch and four new RBAC rules for the snapshot-controller: volumesnapshotcontents:patch, volumesnapshotcontents/status:patch, volumesnapshots:patch, and volumesnapshots/status: patch. (https://github.com/kubernetes-csi/external-snapshotter/pull/526, @ggriffiths)

There is also:

> Rename KUBE_NODE_NAME to NODE_NAME for CSI snapshotter sidecar deployment. (https://github.com/kubernetes-csi/external-snapshotter/pull/616, @zhucan)

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
@sathieu Thanks for the PR can you also update the RBAC changes required as per changelog https://github.com/kubernetes-csi/external-snapshotter/blob/master/CHANGELOG/CHANGELOG-5.0.md#other-cleanup-or-flake and also update the version at https://github.com/rook/rook/blob/master/tests/framework/utils/snapshot.go#L30?

**Comment by sathieu** [comment]
@Madhu-1 Requested changes done. Back to you!

**Comment by Madhu-1** [comment]
> git diff
changes found by 'make gen-rbac'. please run 'make gen-rbac' locally and update your PR
 M deploy/examples/common.yaml


@sathieu CI failure https://github.com/rook/rook/runs/4962590767?check_suite_focus=true can you please check

File: tests/framework/utils/snapshot.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 const (
 	// snapshotterVersion from which the snapshotcontroller and CRD will be
 	// installed
-	snapshotterVersion = "v4.0.0"
+	snapshotterVersion = "v5.0.1"
```
Comment (Madhu-1) [inline tests/framework/utils/snapshot.go:30]: earlier snapshotter was getting deployed in the default namespace, after 4.0.0 its getting deployed in kube-system namespace, I think this change is still not enough we might need to do a string replace to replace kube-system with default or change https://github.com/rook/rook/blob/c57416b2614dc8fa9d2b34ab7ac718937edd56f6/tests/framework/utils/snapshot.go#L86 to kube-system

Comment (Madhu-1) [reply tests/framework/utils/snapshot.go:30]: @sathieu as you are away and we need this fix on priority I send https://github.com/rook/rook/pull/9665 added you as co-author.

**Comment by sathieu** [comment]
Thanks @Madhu-1

**Pull request closed without merging**